cli/runner/cmd/requests.go

```diff
@@ -34,12 +50,65 @@ func getLangListJSON(server string, endpoint string) (*Langs, error) {
 	}
 
 	var jsonLangs Langs
-	json.Unmarshal(body, &jsonLangs)
-	return &jsonLangs, fmt.Errorf("")
+	decodeErr := json.Unmarshal(body, &jsonLangs)
+	panicCheck(decodeErr)
+
+	return &jsonLangs, nil
+}
+
+// This function is designed to send a POST request to the specified endpoint.
+// The server parameter contains an API host, such as 'http://localhost:8080'
+// The endpoint parameter contains an API endpoint, such as '/api/v1/run'
+// The filepath parameter contains the relative filepath to the source file
+// The explicit parameter tells us whether or not the user would like to
+// have the CLI check the language, or to have the server handle it internally
+func postSourceFile(server string, endpoint string, filepath string, langCheck string) (*StdReturn, error) {
+	source, err := os.ReadFile(filepath)
+	if err != nil {
+		return nil, err
+	}
+
+	body := bytes.NewReader(source)
+	req, err := http.NewRequest("POST", server+endpoint, body)
+	if err != nil {
+		return nil, err
+	}
+
+	//TODO: implement CLI check language logic before adding cookie
```

Comment:
Yeah, I was going to ask something similar about API calls for the languages; maybe we could modify the languages API response to be key-value pairs of the language and it's file extension, because otherwise I might have to have the CLI have a hardcoded list of file extensions to check against when doing implicit language checks.

Reply:
Ah good point. I'll work on adding the API request/response data structures after work and I think we could share some variable or function to retrieve the file extension map like is in [coderunner](https://github.com/camerondurham/runner/blob/0d90b5cb57718bd2d503a844290a2f562417e6d6/engine/coderunner/runner.go#L77-L82). Go doesn't have constant maps ([related SO](https://stackoverflow.com/a/18342312/4676641)) sadly so I'll have to use an exported variable or function I think...

I'm not opposed to the API response idea but I think we can just share the same data structure between server and CLI.